charts/karpenter/templates/controller/deployment.yaml

```diff
@@ -1,60 +1,33 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: karpenter-webhook-service
+  name: karpenter-metrics
   namespace: {{ .Release.Namespace }}
-  labels:
-    control-plane: karpenter
 spec:
   ports:
-    - port: 443
-      targetPort: webhook
-  selector:
-    control-plane: karpenter
----
-apiVersion: v1
-kind: Service
-metadata:
-  name: karpenter-metrics-service
-  namespace: {{ .Release.Namespace }}
-  labels:
-    control-plane: karpenter
-spec:
-  ports:
-    - name: http
-      port: 8080
+    - port: 8080
       targetPort: metrics
   selector:
-    control-plane: karpenter
----
-apiVersion: scheduling.k8s.io/v1
-kind: PriorityClass
-metadata:
-  name: karpenter-high-priority
-value: 1000000
-globalDefault: false
-
+    karpenter: controller
 ---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: karpenter
+  name: karpenter-controller
   namespace: {{ .Release.Namespace }}
-  labels:
-    control-plane: karpenter
 spec:
   replicas: 1
   strategy:
     type: Recreate
   selector:
     matchLabels:
-      control-plane: karpenter
+      karpenter: controller
   template:
     metadata:
       labels:
-        control-plane: karpenter
+        karpenter: controller
     spec:
-      priorityClassName: karpenter-high-priority
+      priorityClassName: system-cluster-critical
```

Comment:
In general I agree, and thanks for calling this out. However, I think Karpenter is one of the rare addons (unlike prometheus, etc) which is actually cluster-critical (https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/), since users will rely on it for capacity management. In that sense, I think it shares a similar criticality to the kube-scheduler. 

There is still system-node-critical which is even higher, and is reserved for kube-proxy and aws-node pods.

Reply:
Fair enough. I think that's a good argument.